identity/credentials.go

```diff
@@ -154,6 +166,12 @@ type Credentials struct {
 	// Identifiers represents a list of unique identifiers this credential type matches.
 	Identifiers []string `json:"identifiers" db:"-"`
 
+	// IdentifierAddressType represents the type of the identifiers (e.g. email, phone).
+	// This is used to determine the correct courier to send messages to.
+	// The value is set by the code extension schema and is not persisted.
+	// only applicable on the login, registration with `code` method.
+	IdentifierAddressType CredentialsIdentifierAddressType `json:"-" db:"-"`
```

Comment:
Make it private then maybe and potentially expose it as a function?

Since this is essentially a cache, how can we make sure that the places that needs this always have it (and if not know how to look it up?)

Reply:
It's populated when the identity credentials are verified against the schema. 

I agree, I will move it to a function that exposes it, rather than a property. This value will by default also be "none".